lib/pbench/test/unit/agent/task/test_tools_kill.py

```diff
@@ -0,0 +1,440 @@
+        will all operate normally with no exceptions.  Of the other two, one
+        will raise an expected exception, and the other will not be found.
+        """
+        action_list = []
+        uuid_list = ["uuid1abc", "uuid2def", "uuid3ghi"]
+
+        class MockProcess:
+            """Behavior mock for the `psutil.Process` class."""
+
+            def __init__(
+                self,
+                pid: int,
+                uuid: Optional[str] = None,
+                children: Optional[List[Any]] = None,
+                do: Optional[str] = None,
+            ):
+                self.pid = pid
+                self._uuid = uuid
+                self._children = children
+                self._do = do
+
+            def children(self, recursive: Optional[bool] = False):
+                assert recursive
+                if not self._children:
+                    return
+                for child in self._children:
+                    yield child
+
+            def cmdline(self):
+                return [
+                    "arg0",
+                    "arg1",
+                    "arg2" if not self._uuid else self._uuid,
+                    "arg3",
+                ]
+
+            def kill(self):
+                if self._do == "nosuch":
+                    action_list.append(("kill-nosuch", self.pid, self._uuid))
+                    raise psutil.NoSuchProcess(self.pid)
+                if self._do == "exc":
+                    action_list.append(("kill-except", self.pid, self._uuid))
+                    raise Exception("generic exception")
+                action_list.append(("kill", self.pid, self._uuid))
+
+        class mock_psutil:
+            """Even though the target `psutil` is a module, we just use a
+            class with a static method to duck-type it.
+            """
+
+            @staticmethod
+            def process_iter():
+                # First pid has 3 children
+                yield MockProcess(
+                    12345,
+                    uuid=uuid_list[0],
+                    children=[
+                        MockProcess(12346),
+                        MockProcess(12347),
+                        MockProcess(12348, do="nosuch"),
+                    ],
+                )
+                yield MockProcess(23456)
+                yield MockProcess(34567, uuid=uuid_list[1], do="exc")
+                yield MockProcess(45678, uuid=uuid_list[2], do="nosuch")
+
+        runner = CliRunner(mix_stderr=False)
+        monkeypatch.setattr("pbench.cli.agent.commands.tools.kill.psutil", mock_psutil)
+        result = runner.invoke(kill.main, uuid_list)
+        assert result.exit_code == 0
+        expected_action_list = [
+            ("kill", 12345, "uuid1abc"),
+            ("kill", 12346, None),
+            ("kill", 12347, None),
+            ("kill-nosuch", 12348, None),
+            ("kill-except", 34567, "uuid2def"),
+            ("kill-nosuch", 45678, "uuid3ghi"),
+        ]
```

Comment:
Shouldn't the third value in the tuples at lines 274, 278, and 279 be referencing `uuid_list` (a la lines 258, 266, and 267)?